homework31/main.js

```diff
@@ -77,20 +74,18 @@ const productsById = products.reduce((prev, item) => {
 
 }, {});
 
-
 const productsByCategory = products.reduce((prev, item) => {
     if (!Array.isArray(prev[item.category])) {
         prev[item.category] = [];
     }
-
     prev[item.category].push(item);
+    
     return prev;
-
 }, {});
 
 const categoriesList = Object.keys(productsByCategory);
-let selectedCategory = categoriesList[0];
-let selectedProduct = productsByCategory[selectedCategory][0].id;
+let selectedCategory = 0;
+let selectedProduct = 0;
```

Comment:
Define as `null` or `undefined`, because `0` could be an id of the first item from data
```
let selectedCategory = null;
let selectedProduct = null;

or

let selectedCategory;
let selectedProduct;
```

Then in the code in condition check does it have some value or not
```
- Wrong approach
if (selectedCategory !== 0) {
  // ...
}

- Good approach
if (selectedCategory != null) { // here check not null and not undefined
  // ...
}

```